test/test_parser.cpp

```diff
@@ -5801,8 +5802,9 @@ TEST(Parser_Geospatial)
     CHECK_THROW_EX(
         verify_query(test_context, table, "self_link geoWithin geoBox([0.2, 0.2], [0.7, 0.7])", 0),
         std::runtime_error,
-        CHECK(std::string(e.what()).find("Query 'self_link GEOWITHIN GeoBox([0.2, 0.2], [0.7, 0.7])' links to data "
-                                         "in the wrong format for a geoWithin query") != std::string::npos));
+        CHECK(std::string(e.what()).find(
+                  "Query 'self_link GEOWITHIN GeoPolygon({[0.2, 0.2], [0.2, 0.7], [0.7, 0.7], [0.7, 0.2], [0.2, "
```

Comment:
Also unexpected to get GeoPolygon after geoBox. The 'get_description' would also return transformed query. I'm not sure that is what we want.

Reply:
I think it is best for us to always serialize a box to the equivalent polygon. This will be the text that is sent to the server in the FLX queries, and controlling the conversion of box->polygon on the client ensures that the implementations do not diverge between client and server. (Since MongoDB has no support for queries on a box, they would have to convert to a polygon in the exact same way as the client)

Reply:
By that logic i think it can be argued that we should simply remove GeoBox type all together and simply provide some static initialisation method like "GeoPolygon::from_geobox" as with "GeoCircle::from_kms" for instance.

But still then there is RQL.. GeoBox is in it, and the query changes for local also. We don't send rql to mongo, do we? Wouldn't this be beneficial to hide this implementation detail in local queries, and only really convert to polygon on S2Polygon instantiation for local queries and when the MQL is constructed? 

Still i'm unsure how the user code will handle in the future this implicit conversion of box to polygon. Geospatial type will be constructed from GeoBox, and then saved locally, synced and retrieved later as GeoPolygon. User must be aware of such peculiarity. To me such change of type feels unsettling. Do we have some other precedent like this somewhere else in our api?

Reply:
Yes, GeoPolygon::from_geobox seems like a reasonable API to me. Then we are clear that we don't actually store a GeoBox. Perhaps this is the best way to go because even in phase 2, MongoDB will not support "GeoBox" as a valid geoJSON type?
And yes, the serialized RQL string is being sent to MongoDB for FLX queries.
I can see the incongruence in the API. What do you think about always constructing Polygons  via `GeoPolygon::from_geobox` and never storing a GeoBox in a Geospatial? But as a convenience we can allow `Geospatial::get<GeoBox>()` from a polygon if it has a compatible format for a box?

Reply:
That is essentially the same question we've discussed earlier in some previous pr about this api. GeoCircle is also partially such type - it's not stored and only useful for queries. It's quite convenient for sdks to expose it still as a separate type.

Get rid off it mightn't not be nicer as with GeoBox in the end. My main concerns are that 'to_polygon' is in public api and it's not clear when to do this conversion for sdk: because you can pass GeoBox and GeoPolygon to Geospatial::contains, and also do 'to_string' and from either one for RQL. Which is the right one? Not clear with current approach. 

Also, about serialised query: you've meant server side of sync not mongo itself, right? Server is not going to support 'GeoBox' operator in RQL, that's what you're saying, right? So it means we have to convert for rql... I'm ok with this change as long as the results with mongo matches. It just makes api quirky in the end.

Would be great to hear some thoughts from sdk perspective @papafe @nirinchev anyone?

Reply:
@kiburtse I just added `GeoBox::from_polygon(const GeoPolygon&)` to the API so that there is a symmetry for boxes. It should produce equivalent results running `Geospatial::contains` on a box or the polygon that it produces. 

> Also, about serialised query: you've meant server side of sync not mongo itself, right? Server is not going to support 'GeoBox' operator in RQL, that's what you're saying, right? So it means we have to convert for rql... I'm ok with this change as long as the results with mongo matches. It just makes api quirky in the end.

Yes that's correct. I don't think they have implmented the server side syntax yet, but it will simplify their project and reduce inconsistencies between the client/server results if the server doesn't have to support a GeoBox in a query at all.